### ISSUE TITLE
on neural replicator dynamics

### ISSUE THREAD

**Issue by linshuxi** [deepmind/open_spiel#921]
Hi guys,
I am trying to implement the neural replicator dynamics algorithm in the paper "Neural Replicator Dynamics:
Multiagent Learning via Hedging Policy Gradients", and the link of the article is https://arxiv.org/pdf/1906.00190.pdf.

The pseudo code is:
![2022-08-29 10-04-34屏幕截图](https://user-images.githubusercontent.com/6730777/187108954-03238248-e51d-4b81-bed9-e73620853ed4.png)

I have a few questions, and any suggestions are appreciated
(1) The critic network outputs q value, not v value. Is it correct?  Is there any benefits for using q value? Because the critic network outputs v value in A2C,so i think v value is more reasonable.

(2) The algorithm first sample some trajectories and then do the policy evaluation, which seems like monte carlo sampling. Is it right?
It uses TD error in A2C to do the update. Is it a good idea to use TD error to do the policy evaluation?

(3) A2C and neural RD is very similar, and they both have a critic network and a policy network. There is only one difference, and the last softmax layer is cut off in the neural RD. If i am right, can i just cut off the last softmax layer in A2C and then train the rest part of the network?

**Comment by linshuxi** [comment]
Hi @lanctot, thanks a lot for the response, and it is very useful. I will share the results when i finish the experiments.
